.Rprofile

```diff
@@ -1,45 +1,65 @@
 # source global .Rprofile (very important to load user specific settings)
 # DO NOT EDIT THIS LINE!
-if(file.exists("~/.Rprofile")) source("~/.Rprofile")
+if (file.exists("~/.Rprofile")) {
+  source("~/.Rprofile")
+}
 
 # This profile can be used to link the model to a specified library snapshot
 # (e.g. if your model version is from an older date and does not work with the
 # newest libraries anymore). By default it is not active.
 
+local({ # prevent vairables defined here from ending up in the global env
+
+# snapshots created with R 4.1 have a _R4 postfix
+maybeR4 <- if (R.version$major == 4) "_R4" else ""
 # Gather all library snapshots
-snapshots <- sort(list.files(
-             path = '/p/projects/rd3mod/R/libraries/snapshots',
-             pattern = '^20[0-9]{2}_[0-9]{2}(_[0-9]{2})?$',
-             full.names = TRUE), decreasing = TRUE)
-latest_snapshot <- ifelse (length(snapshots) > 0, snapshots[[1]], NA)
+snapshots <- sort(list.files(path = '/p/projects/rd3mod/R/libraries/snapshots',
+                             pattern = paste0('^20[0-9]{2}_[0-9]{2}(_[0-9]{2})?', maybeR4, '$'),
+                             full.names = TRUE), decreasing = TRUE)
+latestSnapshot <- if (length(snapshots) > 0) snapshots[[1]] else NA
+
+activateSnapshot <- function(snapshot) {
```

Comment:
Wouldn't it be clever to check here again whether the snapshot contains or does not contain `_R4`, depending on `R.version$major`? Because certainly someone (me) will try to specify a snapshot manually that doesn't fit the R version and I would prefer that the script checks this.